aten/src/ATen/native/cudnn/MHA.cpp

```diff
@@ -331,40 +377,19 @@ auto build_graph_and_tensors(
               std::vector<int64_t>(
                   v.strides().data(),
                   v.strides().data() + v.strides().size()))));
-  auto attn_scale =
-      mha_graph->tensor(fe::graph::Tensor_attributes()
-                            .set_name("Attn_scale")
-                            .set_dim({1, 1, 1, 1})
-                            .set_stride({1, 1, 1, 1})
-                            .set_is_pass_by_value(true)
-                            .set_data_type(fe::DataType_t::FLOAT));
-  // TODO(eqy): support bias in the future in a follow-up PR
-  // auto bias = mha_graph->tensor(fe::graph::Tensor_attributes()
-  //                         .set_name("bias")
-  //                         .set_dim({b, 1, s_q, s_kv})
-  //                         .set_stride({s_q * s_kv, s_q * s_kv, s_kv, 1}));
-  auto seed = mha_graph->tensor(fe::graph::Tensor_attributes()
-                                    .set_name("Seed")
-                                    .set_dim({1, 1, 1, 1})
-                                    .set_stride({1, 1, 1, 1})
-                                    .set_data_type(fe::DataType_t::INT32));
-  auto offset = mha_graph->tensor(fe::graph::Tensor_attributes()
-                                      .set_name("Offset")
-                                      .set_dim({1, 1, 1, 1})
-                                      .set_stride({1, 1, 1, 1})
-                                      .set_data_type(fe::DataType_t::INT32));
-  auto scaled_dot_product_flash_attention_options =
-      fe::graph::SDPA_attributes()
-          .set_name("CUDNN_SDPA")
-          .set_is_inference(return_softmaxstats == false)
-          .set_causal_mask(is_causal)
-          .set_attn_scale(attn_scale)
-          .set_dropout(dropout_probability, seed, offset);
-  // Optional bias in flash attention is only supported 8.9.3 onwards
-  if (cudnnGetVersion() >= 8904) {
-    // scaled_dot_product_flash_attention_options.set_alibi_mask(true);
+  std::optional<std::shared_ptr<fe::graph::Tensor_attributes>> bias;
+  if (attn_bias.has_value()) {
+    bias = mha_graph->tensor(fe::graph::Tensor_attributes()
+                                 .set_name("bias")
+                                 .set_dim(std::vector<int64_t>(
```

Comment:
I was taking a look at the call chain for set_dim and set_stride. Its really a bummer that there isnt an rvalue overload for set_dim so that we only have to do one clone instead of 2 like it is today. that being said does

``` cpp
    bias = mha_graph->tensor(fe::graph::Tensor_attributes()
                                 .set_name("bias")
                                 .set_dim(attn_bias.value().sizes().vec())
                                 .set_stride(attn_bias.value().strides().vec()));                  
  ````
  work? I think its alot more readable but this is a nit